### ISSUE TITLE
template variables in url param expand to all values instead of current value

### ISSUE THREAD

**Issue by dsvensson** [grafana/grafana#14367]
### What Grafana version are you using?
5.3.2 and 5.4
### What datasource are you using?
PostgreSQL
### What OS are you running grafana on?
Linux
### What did you do?
Generate a number of singlestat-panels using a template variable with 'Include All' checkbox checked, to have it iterate over all the values and create panels, each panel having $thevariable as both Title, but also as URL-param in a Dashboard link.
### What was the expected result?
When clicking a singlestat-panel, I expect to go to the target dashboard, with an url-parameter corresponding to the singlestat-panel variable expansion.
### What happened instead?
The variable in the url-param is expanded to all possible values of the template variable. Yet the title is correctly named on each panel, using the same variable as in the URL-param. Another weird thing is that the variable is, in the URL, set to `var-server={'value1','value2'}` rather than expanding to multiple instances of the url param like what happens when you select a multi-variable.

**Comment by torkelo** [comment]
unable to replicate this, can you try v5.4? 

I tried with both manually specifying the variable and checking the "include variable" checkbox. When I click on a singlestat panel it opens the dashboard with the correct url / variable value set. 

![image](https://user-images.githubusercontent.com/10999/49634880-43d76880-f9fe-11e8-8379-bf4a78c57813.png)

**Comment by dsvensson** [comment]
@torkelo Nopes, just upgraded to 5.4, still broken. Looks like you're not using Repeat in your screenshot. Maybe that's it. In your sample, if test would have `['a','b','c']`, and All expanding those into three panels, that url param in each one would result in `&var-test={'a','b','c'}`, at least here.

**Comment by torkelo** [comment]
no, I have a repeat enabled, just took screenshot from one the repeated panels instead of the source panel by accident.

**Comment by dsvensson** [comment]
@torkelo Then you need to elaborate on what details you need. The 'a' panel does not result in `&var-test=a` here like you proclaim, instead it includes all, within encoded {}, which are unusable by the selection box on the other dashboard.

**Comment by torkelo** [comment]
Can you test with this dashboard: 

```
{
  "__inputs": [
    {
      "name": "DS_GDEV-TESTDATA",
      "label": "gdev-testdata",
      "description": "",
      "type": "datasource",
      "pluginId": "testdata",
      "pluginName": "TestData DB"
    }
  ],
  "__requires": [
    {
      "type": "grafana",
      "id": "grafana",
      "name": "Grafana",
      "version": "5.5.0-pre1"
    },
    {
      "type": "panel",
      "id": "singlestat",
      "name": "Singlestat",
      "version": "5.0.0"
    },
    {
      "type": "datasource",
      "id": "testdata",
      "name": "TestData DB",
      "version": "1.0.0"
    }
  ],
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": "-- Grafana --",
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "gnetId": null,
  "graphTooltip": 0,
  "id": null,
  "iteration": 1544173146148,
  "links": [],
  "panels": [
    {
      "cacheTimeout": null,
      "colorBackground": false,
      "colorValue": false,
      "colors": [
        "#299c46",
        "rgba(237, 129, 40, 0.89)",
        "#d44a3a"
      ],
      "datasource": "${DS_GDEV-TESTDATA}",
      "format": "none",
      "gauge": {
        "maxValue": 100,
        "minValue": 0,
        "show": false,
        "thresholdLabels": false,
        "thresholdMarkers": true
      },
      "gridPos": {
        "h": 9,
        "w": 4.8,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "interval": null,
      "links": [
        {
          "dashboard": "Single graph render",
          "includeVars": true,
          "params": "",
          "title": "Single graph render",
          "type": "dashboard",
          "url": "/d/JgptJFBmz/single-graph-render"
        }
      ],
      "mappingType": 1,
      "mappingTypes": [
        {
          "name": "value to text",
          "value": 1
        },
        {
          "name": "range to text",
          "value": 2
        }
      ],
      "maxDataPoints": 100,
      "minSpan": 3,
      "nullPointMode": "connected",
      "nullText": null,
      "postfix": "",
      "postfixFontSize": "50%",
      "prefix": "",
      "prefixFontSize": "50%",
      "rangeMaps": [
        {
          "from": "null",
          "text": "N/A",
          "to": "null"
        }
      ],
      "repeat": "test",
      "repeatDirection": "h",
      "sparkline": {
        "fillColor": "rgba(31, 118, 189, 0.18)",
        "full": false,
        "lineColor": "rgb(31, 120, 193)",
        "show": false
      },
      "tableColumn": "",
      "targets": [
        {
          "expr": "",
          "format": "time_series",
          "intervalFactor": 1,
          "refId": "A",
          "scenarioId": "random_walk"
        }
      ],
      "thresholds": "",
      "title": "Panel Title",
      "type": "singlestat",
      "valueFontSize": "80%",
      "valueMaps": [
        {
          "op": "=",
          "text": "N/A",
          "value": "null"
        }
      ],
      "valueName": "avg"
    }
  ],
  "schemaVersion": 16,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": [
      {
        "allValue": null,
        "current": {
          "tags": [],
          "text": "All",
          "value": [
            "$__all"
          ]
        },
        "hide": 0,
        "includeAll": true,
        "label": null,
        "multi": true,
        "name": "test",
        "options": [
          {
            "selected": true,
            "text": "All",
            "value": "$__all"
          },
          {
            "selected": false,
            "text": "1",
            "value": "1"
          },
          {
            "selected": false,
            "text": "2",
            "value": "2"
          },
          {
            "selected": false,
            "text": "3",
            "value": "3"
          },
          {
            "selected": false,
            "text": "4",
            "value": "4"
          },
          {
            "selected": false,
            "text": "5",
            "value": "5"
          }
        ],
        "query": "1,2,3,4,5",
        "skipUrlSync": false,
        "type": "custom"
      }
    ]
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {
    "refresh_intervals": [
      "5s",
      "10s",
      "30s",
      "1m",
      "5m",
      "15m",
      "30m",
      "1h",
      "2h",
      "1d"
    ],
    "time_options": [
      "5m",
      "15m",
      "1h",
      "6h",
      "12h",
      "24h",
      "2d",
      "7d",
      "30d"
    ]
  },
  "timezone": "",
  "title": "repeat links",
  "uid": "-JYqb4Pik",
  "version": 4
}
```

you will need to add a data source of the type "Test Data"

**Comment by dsvensson** [comment]
@torkelo According to the sysmin there seems to be some bug that prevents enabling the Test Data datasource. Is there any other way to give you more data?

**Comment by dsvensson** [comment]
@torkelo I've experimented a bit more. If you configure the datasource to a postgresql datasource in the metrics tab in your singlestat panels, you should get my behavior.

**Comment by dsvensson** [comment]
@torkelo Did you manage to reproduce it when switching the data source in the singlestats-metrics-tab to postgresql?

**Comment by dsvensson** [comment]
@torkelo Just did another test. Started a fresh 5.4 in docker, imported your template. Verified that each link leads to var-test=1, var-test=2 etc... switched the singlestat metrics datasource to a postgresql, and now they all result in var-test=All url param... not the {1,2,3,4} as I expected, but still just as wrong.

**Comment by torkelo** [comment]
thanks for adding additional info, I was now able to replicate the issue if I use postgres query that returns table data (instead of time series data). 

Fix incoming in v5.4.1 
https://github.com/grafana/grafana/pull/14426
